spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core;
+
+import java.util.List;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.Options;
+import com.google.cloud.spanner.Statement;
+
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerMappingContext;
+
+/**
+ * Defines operations available to use with Spanner.
+ *
+ * @author Ray Tsang
+ * @author Chengyuan Zhao
+ */
+public interface SpannerOperations {
+
+	/**
+	 * Gets the mapping context used to facilitate the available options.
+	 * @return the mapping context.
+	 */
+	SpannerMappingContext getMappingContext();
```

Comment:
Why is this method required? Can we introduce it when there are other users then the tests?

Reply:
Yes I can remove this one. 

I'm fine with doing the "thin slice" approach, and I think the simple methods mentioned in this PR constitute thin enough of a slice. They're all pretty similar and will share lots of implementation logic. I don't think we should be slicing this particular PR any thinner.

Reply:
Okay. I guess this "slice" can be borderline accepted size-wise, and I don't want you to revert all your changes artificially. The rationale behind going even more thinner slices, i.e. babysteps because that creates even smaller PRs (<<<700 lines, <<13 methods), so that people can review them quickly, also gives a chance to think about every single detail. 

Can you please provide the implementation for the other interfaces as well then, so that we have an end-to-end working SpannerTemplate?

Reply:
We initially discussed getting the template up here and that is this interface here. I will add the an implementation in the very next PR, because that and its unit tests together will make this PR swell even more.